app/src/ui/branches/pull-request-list.tsx

```diff
@@ -48,19 +51,33 @@ export class PullRequestList extends React.Component<
   public constructor(props: IPullRequestListProps) {
     super(props)
 
+    const group = createListItems(props.pullRequests)
+    const pullRequest = props.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
+      : null
+
     this.state = {
-      groupedItems: [createListItems(props.pullRequests)],
+      groupedItems: [group],
       filterText: '',
-      selectedItem: null,
+      selectedItem: selectedItem || null,
     }
   }
 
   public componentWillReceiveProps(nextProps: IPullRequestListProps) {
-    if (nextProps.pullRequests === this.props.pullRequests) {
+    if (
+      nextProps.pullRequests === this.props.pullRequests &&
+      nextProps.currentPullRequest === this.props.currentPullRequest
+    ) {
       return
     }
 
-    this.setState({ groupedItems: [createListItems(nextProps.pullRequests)] })
+    const group = createListItems(nextProps.pullRequests)
+    const pullRequest = nextProps.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
+      : this.state.selectedItem
```

Comment:
Can we really trust `this.state.selectedItem` to exist in the newly created list? It seems we'd need to do a lookup here comparing on `.id` as well, no?